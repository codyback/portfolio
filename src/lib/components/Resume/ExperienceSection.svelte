<script lang="ts">
	import type {
		Experience,
		ExperienceResponsibility,
		ExperienceProject,
	} from '@prisma/client';
	import { getExperienceDateRange } from '@/lib/utils/DateUtils';

	type ExperienceWithRelations = Experience & {
		responsibilities: ExperienceResponsibility[];
		projects: ExperienceProject[];
	};

	export let experiences: ExperienceWithRelations[];
</script>

<section id="cvExperiences">
	<div class="mt-10 sm:mt-20 border-b-2 border-black pb-3">
		<h3 class="uppercase text-2xl font-bold text-right">Experience</h3>
	</div>

	<div class="border-t-4 border-black mt-0.5 pt-8">
		{#each experiences as experience, i}
			<div>
				<div class="sm:grid sm:grid-cols-2 items-center font-bold">
					<div class="sm:col-span-1 text-lg leading-none">
						{experience.company}
					</div>
					<div class="sm:col-span-1 sm:text-right sm:pt-0 pt-4">
						{getExperienceDateRange(experience.startDate, experience.endDate)}
						<br />
						{experience.position}
					</div>
				</div>

				{#if experience.description}
					<p class="pt-4 text-neutral-500">
						{experience.description}
					</p>
				{/if}

				{#if experience.responsibilities.length !== 0}
					<ul class="pt-4 list-disc text-neutral-500">
						Responsibilities include:
						{#each experience.responsibilities as responsibility}
							<li class="ml-4">
								{responsibility.description}
							</li>
						{/each}
					</ul>
				{/if}

				{#if experience.projects.length !== 0}
					<ul class="pt-4 list-disc text-neutral-500">
						Projects worked on:
						{#each experience.projects as project}
							<li class="ml-4">
								{#if project.url}
									<a
										href={project.url}
										class="hover:text-black underline font-bold"
										>{project.name}</a
									>: {project.description}
								{:else}
									<span class="font-bold">{project.name}</span>: {project.description}
								{/if}
							</li>
						{/each}
					</ul>
				{/if}
			</div>
			{#if i + 1 !== experiences.length}
				<hr class="my-8" />
			{/if}
		{/each}
	</div>
</section>
