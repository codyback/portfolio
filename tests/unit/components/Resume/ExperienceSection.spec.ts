import ExperienceSection from '$lib/components/Resume/ExperienceSection.svelte';
import type {
  Experience,
  ExperienceResponsibility,
  ExperienceProject,
} from '@prisma/client';
import { render } from '@testing-library/svelte';
import { describe, it, expect } from 'vitest';

type ExperienceWithRelations = Experience & {
  responsibilities: ExperienceResponsibility[];
  projects: ExperienceProject[];
};

const experiences: ExperienceWithRelations[] = [
  {
    id: 'test',
    company: 'Testing Company 1',
    url: 'url-here',
    description:
      'Testing Company 1 description. Adipisicing amet aliquip occaecat Lorem non do laboris qui consectetur proident ex occaecat.',
    position: 'Testing Position 1',
    startDate: new Date('01-01-2000'),
    endDate: null,
    isEmployed: true,
    homepageId: 'test-1',
    responsibilities: [
      {
        id: 'resp-1',
        description: 'resp desc 1',
        experienceId: 'test',
      },
      {
        id: 'resp-2',
        description: 'resp desc 2',
        experienceId: 'test',
      },
    ],
    projects: [
      {
        id: 'proj-1',
        name: 'Project 1',
        url: 'proj-1-url',
        description: 'proj-1 desc',
        experienceId: 'test',
      },
      {
        id: 'proj-2',
        name: 'Project 2',
        url: 'proj-2-url',
        description: 'proj-2 desc',
        experienceId: 'test',
      },
    ],
  },
  {
    id: 'test-2',
    company: 'Testing Company 2',
    url: 'url-here',
    description:
      'Testing Company 2 description. Aliqua aute sint exercitation exercitation excepteur enim excepteur duis eiusmod occaecat voluptate aliquip reprehenderit.',
    position: 'Testing Position 2',
    startDate: new Date('01-01-2000'),
    endDate: new Date('01-01-2000'),
    isEmployed: false,
    homepageId: 'test-1',
    responsibilities: [],
    projects: [],
  },
];

describe('CertificatesSection', () => {
  it('renders correctly with open false', () => {
    const { container } = render(ExperienceSection, { experiences });

    expect(container).toMatchSnapshot();
  });
});
