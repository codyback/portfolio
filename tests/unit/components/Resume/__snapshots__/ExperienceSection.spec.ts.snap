// Vitest Snapshot v1

exports[`CertificatesSection > renders correctly with open false 1`] = `
<body>
  <div>
    <section
      class="mt-10 lg:mt-20"
      id="cvExperiences"
    >
      <div
        class="border-b-2 border-black dark:border-neutral-500 pb-3"
      >
        <h3
          class="uppercase text-2xl font-bold text-right"
        >
          Experience
        </h3>
      </div>
       
      <div
        class="border-t-4 border-black dark:border-neutral-500 mt-0.5"
      />
      <!--&lt;SectionHeading&gt;-->
       
      <div
        class="pt-8"
      >
        <div>
          <div
            class="sm:grid sm:grid-cols-2 items-center font-bold"
          >
            <div
              class="sm:col-span-1 text-lg leading-none"
            >
              Testing Company 1
            </div>
             
            <div
              class="sm:col-span-1 sm:text-right sm:pt-0 pt-4"
            >
              2000 - Current
               
              <br />
               
              Testing Position 1
            </div>
          </div>
           
          <p
            class="pt-4 text-neutral-500 dark:text-neutral-300"
          >
            Testing Company 1 description. Adipisicing amet aliquip occaecat Lorem non do laboris qui consectetur proident ex occaecat.
          </p>
           
          <ul
            class="pt-4 list-disc text-neutral-500 dark:text-neutral-300"
          >
            Responsibilities include:
						
            <li
              class="ml-4"
            >
              resp desc 1
               
            </li>
            <li
              class="ml-4"
            >
              resp desc 2
               
            </li>
          </ul>
           
          <ul
            class="pt-4 list-disc text-neutral-500 dark:text-neutral-300"
          >
            Projects worked on:
						
            <li
              class="ml-4"
            >
              <a
                class="hover:text-black dark:hover:text-white underline font-bold"
                href="proj-1-url"
              >
                Project 1
              </a>
              : 
              proj-1 desc
               
            </li>
            <li
              class="ml-4"
            >
              <a
                class="hover:text-black dark:hover:text-white underline font-bold"
                href="proj-2-url"
              >
                Project 2
              </a>
              : 
              proj-2 desc
               
            </li>
          </ul>
        </div>
         
        <hr
          class="my-8"
        />
        
        <div>
          <div
            class="sm:grid sm:grid-cols-2 items-center font-bold"
          >
            <div
              class="sm:col-span-1 text-lg leading-none"
            >
              Testing Company 2
            </div>
             
            <div
              class="sm:col-span-1 sm:text-right sm:pt-0 pt-4"
            >
              2000 - 2000
               
              <br />
               
              Testing Position 2
            </div>
          </div>
           
          <p
            class="pt-4 text-neutral-500 dark:text-neutral-300"
          >
            Testing Company 2 description. Aliqua aute sint exercitation exercitation excepteur enim excepteur duis eiusmod occaecat voluptate aliquip reprehenderit.
          </p>
           
           
        </div>
         
        
      </div>
    </section>
    <!--&lt;ExperienceSection&gt;-->
  </div>
</body>
`;
